### PR TITLE
MOTECH-2054 Replace LoadOnDemand.js

### PR DESCRIPTION
**Pull request by wstrzelczyk**
MOTECH-2054 Replace LoadOnDemand.js

### CODE REVIEW

File: modules/admin/src/main/resources/webapp/js/app.js [hunk 1]
```diff
@@ -19,5 +142,5 @@
               when('/admin/queues/browse', {templateUrl: '../admin/partials/queue_message_stats.html', controller: 'AdminQueueMessageStatsCtrl'}).
               when('/admin/logOptions', {templateUrl: '../admin/partials/logOptions.html', controller: 'AdminServerLogOptionsCtrl'}).
               when('/admin/messagesSettings', {templateUrl: '../admin/partials/notificationRules.html', controller: 'AdminNotificationRuleCtrl'});
-    }]);
+    }]);*/
```
Comment (pgesek) [inline modules/admin/src/main/resources/webapp/js/app.js]: Commented out code

File: modules/scheduler/scheduler/src/main/resources/webapp/js/app.js [hunk 2]
```diff
@@ -7,7 +7,42 @@
...
+                views: {
+                    'schedulerview': {
+                        templateUrl: '../scheduler/partials/createJob.html',
+                        controller: 'SchedulerCreateJobCtrl'
+                    }
+                }
+            });
+    });
+
+        /*scheduler.config( function ($routeProvider) {
```
Comment (pgesek) [inline modules/scheduler/scheduler/src/main/resources/webapp/js/app.js]: commented out code

File: modules/tasks/tasks/src/main/resources/webapp/js/app.js [hunk 3]
```diff
@@ -5,13 +5,67 @@
...
+                    }
+                }
+            });
+               /* when('/tasks/dashboard', {templateUrl: '../tasks/partials/tasks.html', controller: 'TasksDashboardCtrl'}).
                 when('/tasks/task/new', {templateUrl: '../tasks/partials/form.html', controller: 'TasksManageCtrl'}).
                 when('/tasks/task/:taskId/edit', {templateUrl: '../tasks/partials/form.html', controller: 'TasksManageCtrl'}).
                 when('/tasks/task/:taskId/log', {templateUrl: '../tasks/partials/history.html', controller: 'TasksLogCtrl'}).
-                when('/tasks/settings', {templateUrl: '../tasks/partials/settings.html', controller: 'TasksSettingsCtrl'});
-        }]);
+                when('/tasks/settings', {templateUrl: '../tasks/partials/settings.html', controller: 'TasksSettingsCtrl'});*/
```
Comment (pgesek) [inline modules/tasks/tasks/src/main/resources/webapp/js/app.js]: commented out code

File: platform/server-bundle/src/main/resources/webapp/WEB-INF/jsp/man.html [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<div>
```
Comment (pgesek) [inline platform/server-bundle/src/main/resources/webapp/WEB-INF/jsp/man.html]: what is this?

File: platform/server-bundle/src/main/resources/webapp/js/dashboard.js [hunk 5]
```diff
@@ -1,4 +1,4 @@
-function initAngular() {
+/*function initAngular() {
```
Comment (pgesek) [inline platform/server-bundle/src/main/resources/webapp/js/dashboard.js]: commeneted out code, do we need this file?

**Comment by pgesek** [comment]
@aczyrko test the following:

- module menus
- transitioning between modules
- loading tab UI in module
- language change
- embedded MDS views - for example in IVR or Message Campaign
- if the UI is working in general (i.e. just creating a task, using the mds data browser etc.)

**Comment by sebbrudzinski** [comment]
@wstrzelczyk The MOTECH archetypes must also be updated to work with LoadOnDemand library

**Comment by sebbrudzinski** [comment]
Conflicts must be resolved.

**Comment by aczyrko** [comment]
Errors:
- When I click Admin → Queues I get 500 Internal server error.
- When I click Admin → Topics I get 500 Internal server error.
- When I double click in Metrics in left side menu appears blocking screen (application frozen) and appears error in console.
- When I click in the left side menu in SMS in console I have a message „no element found”.

More details in attachment.
[PR474.pdf](https://github.com/motech/motech/files/231423/PR474.pdf)

**Comment by wstrzelczyk** [comment]
I fixed Metrics module and Server local time, Server upTime.
For the rest of errors please create new tickets, they are not connected with my changes.

**Comment by sebbrudzinski** [comment]
FYI, @aczyrko

**Comment by aczyrko** [comment]
Server local time, Server upTime works

**Errors**
* When I switched module for Metrix previous error do not appear but two new 500 errors appear.
* When I want to add configuration in Commcare I get error in console.
* When I click ODK in menu i get 500 error.

More information in attachment: 
[PR474 version 29.04.2016.pdf]

**Comment by wstrzelczyk** [comment]
@aczyrko All reported errors are not connected with my changes, please create new tickets if not exist.

**Comment by sebbrudzinski** [comment]
@aczyrko @pgesek Are we good to merge this?

**Comment by pgesek** [comment]
What's wrong with Commcare?

**Comment by pgesek** [comment]
And Metrics and ODK for that matter, but I'm primarily concerned with the Commcare issue

**Comment by sebbrudzinski** [comment]
@pgesek There seems to be a problem retrieving the confoguration from the server for Commcare, Metrics and SMS, but it does not seem related to this ticket (can be seen on master). ODK also logs some errors to the console, but it doesn't seem to affect the module. I'll go ahead and create tickets for those issues myself.

**Comment by pgesek** [comment]
@sebbrudzinski These are just console issues though right?

**Comment by sebbrudzinski** [comment]
@pgesek No, those are actual flaws, that prevent adding new configs

**Comment by PiotrZurawskiSolDevelo** [comment]
I can not reproduce this on my local master.
Am I missing something?

**Comment by sebbrudzinski** [comment]
It's possible that it is yet another issue, that manifests itself only when certain modules are started.

**Comment by sebbrudzinski** [comment]
@pgesek @PiotrZurawskiSolDevelo 
I've found out the cause of all evil is the Metrics module. Configs in SMS and Commcare work properly if Metrics are not started. See https://applab.atlassian.net/browse/MOTECH-2461

**Comment by PiotrZurawskiSolDevelo** [comment]
I checked this. I confirm that there is error during adding New configuration into commcare module when all modules are installed. Rest of errors also exist.

**Pull request merged**